src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -2,16 +2,26 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.IO;
+using System.Text.Encodings.Web;
 using System.Threading.Tasks;
+using Microsoft.AspNet.Html.Abstractions;
 
 namespace Microsoft.AspNet.Razor.TagHelpers
 {
     /// <summary>
     /// Class used to represent the output of an <see cref="ITagHelper"/>.
     /// </summary>
-    public class TagHelperOutput
+    public class TagHelperOutput : IHtmlContent
     {
         private readonly Func<bool, Task<TagHelperContent>> _getChildContentAsync;
+        private TagHelperContent _preElement;
+        private TagHelperContent _preContent;
+        private TagHelperContent _content;
+        private TagHelperContent _postContent;
+        private TagHelperContent _postElement;
+        private bool _contentSupressed;
```

Comment:
Hmm, "content suppressed" is the state only immediately after a call to `SuppressOutput()`. Suggest `_wasSuppressOutputCalled` especially to avoid confusing @rynowak again.